### PR TITLE
Removing HashedStream from upload

### PR DESCRIPTION
**Pull request by nhibberd**
@markhibberd @charleso

### CODE REVIEW

**Comment by markhibberd** [comment]
Can we add a reliabilty test for upload of small files. After that :+1:

File: mismi-s3/src/Mismi/S3/Patch/Network.hs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import           System.IO
+
+chunkedFile :: MonadIO m => ChunkSize -> FilePath -> m RqBody
+chunkedFile c f = do
+    n <- getFileSize f
+    if n > toInteger c
+      then
+        return $ unsafeChunkedBody c n (sourceFileChunks c f)
+      else
+        liftIO $ (=<<) (return . toBody) $ LBS.readFile f
```
Comment (charleso) [inline mismi-s3/src/Mismi/S3/Patch/Network.hs]: Not sure if it's worth commenting about how/why this line is different?

File: mismi-s3/src/Mismi/S3/Patch/Network.hs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+-- Uses hGet with a specific buffer size, instead of hGetSome.
+sourceFileChunks :: MonadResource m
```
Comment (charleso) [inline mismi-s3/src/Mismi/S3/Patch/Network.hs]: Pity this isn't exported from amazona. :(

**Comment by charleso** [comment]
:+1:

File: mismi-s3/src/Mismi/S3/Patch/Network.hs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                 -> FilePath
+                 -> Source m BS.ByteString
+sourceFileChunks (ChunkSize sz) f =
+  bracketP (openBinaryFile f ReadMode) hClose go
+  where
+    go h = do
+        bs <- liftIO (BS.hGet h sz)
+        unless (BS.null bs) $ do
+            yield bs
+            go h
```
Comment (jystic) [inline mismi-s3/src/Mismi/S3/Patch/Network.hs]: Ohhh, now I see what you were talking about

Comment (jystic) [inline mismi-s3/src/Mismi/S3/Patch/Network.hs]: Or is this different to the question you were asking about before?

**Comment by etorreborre** [comment]
@nhibberd is this to fix #229?

**Comment by nhibberd** [comment]
Yup, fixes #229

**Pull request merged**